pkg/services/live/pipeline/pipeline.go

```diff
@@ -274,3 +343,44 @@ func (p *Pipeline) processFrame(ctx context.Context, orgID int64, channelID stri
 
 	return nil, nil
 }
+
+func (p *Pipeline) processData(ctx context.Context, orgID int64, channelID string, data []byte) ([]*ChannelData, error) {
+	rule, ruleOk, err := p.ruleGetter.Get(orgID, channelID)
```

Comment:
we are extracting rules several times while pipeline works – I think it's possible to refactor to get Rule from outside pipeline and then call pipeline with that rule, and then on redirects get next rules. Would be less repetitive.